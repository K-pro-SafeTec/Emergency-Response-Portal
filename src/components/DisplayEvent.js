import React, { Component } from 'react';
import Paper from '@material-ui/core/Paper';
import Button from '@material-ui/core/Button';


export class DisplayEvent extends Component {
  
  render() {
    const event = this.props.event;
    return (
      <Paper className="paper">
        <h1>{event.type}</h1>
        <h2>{getTimeFormat(event.start)}-{getTimeFormat(event.end)}</h2>
        <h2>Deltakere: {event.participants}</h2>
        <div>
          <Button variant="outlined" color="primary">
            Endre
          </Button>
          <Button variant="outlined" color="secondary">
            Slett
          </Button>
        </div>
      </Paper>
    )
  }
}



// Takes javascript date object and returns nice time
function getTimeFormat(date) {
  let hours = date.getHours();
  let minutes = date.getMinutes();
  if (hours < 10) {
    hours = "0" + hours;
  }
  if (minutes < 10) {
    minutes = "0" + minutes;
  }
  return hours + ":" + minutes;
}