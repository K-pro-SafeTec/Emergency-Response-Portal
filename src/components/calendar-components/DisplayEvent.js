import React, { Component } from 'react';
import Paper from '@material-ui/core/Paper';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';

export class DisplayEvent extends Component {
  
  render() {
    const event = this.props.event;
    return (
      <Paper className="paper-small">
        <Typography variant="headline" component="h2">
          {event.title}
        </Typography>
        <Typography component="p">
          {getTimeFormat(event.start)}-{getTimeFormat(event.end)}
        </Typography>
        <Typography component="p">
          Deltakere: {event.participants}
        </Typography>
        <div>
          <Button style={{marginRight: '5px'}} variant="outlined" color="primary">
            Endre
          </Button>
          <Button variant="outlined" color="secondary">
            Slett
          </Button>
        </div>
      </Paper>
    )
  }
}



// Takes javascript date object and returns nice time
function getTimeFormat(date) {
  let hours = date.getHours();
  let minutes = date.getMinutes();
  if (hours < 10) {
    hours = "0" + hours;
  }
  if (minutes < 10) {
    minutes = "0" + minutes;
  }
  return hours + ":" + minutes;
}