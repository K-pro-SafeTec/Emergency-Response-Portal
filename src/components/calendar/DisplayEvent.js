import React from 'react';
import PropTypes from 'prop-types';
import Paper from '@material-ui/core/Paper';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';
import {getTimeFormat} from './../../helpers/calendar-helper';

const DisplayEvent = (props) => {
  const event = props.event;
  return (
    <Paper className="paper-small">
      <Typography variant="headline" component="h2">
        {event.title}
      </Typography>
      <Typography component="p">
        {getTimeFormat(event.start)}-{getTimeFormat(event.end)}
      </Typography>
      <Typography component="p">
        Deltakere: {event.participants}
      </Typography>
      <Button style={{marginRight: '5px'}} variant="outlined" color="primary" onClick={() => props.onChangeEvent(event.id)}>
        Endre
      </Button>
      <Button href="https://drive.google.com/file/d/1SIy2luM50jY-_od88vfXg5k7l0oT8C0s/view?usp=sharing" target="_blank"
              style={{marginRight: '5px'}} variant="outlined" color="primary"
              onClick={() => props.onReviewButtonClick(event.id)}>
        Se debrief
      </Button>
      <Button variant="outlined" color="secondary" onClick={() => props.onDeleteButtonClick(event.id)}>
        Slett
      </Button>
    </Paper>
  )
};

DisplayEvent.propTypes = {
  event: PropTypes.object.isRequired,
  onChangeEvent: PropTypes.func.isRequired,
  onDeleteButtonClick: PropTypes.func.isRequired,
  onReviewButtonClick: PropTypes.func.isRequired
};

export default DisplayEvent;
