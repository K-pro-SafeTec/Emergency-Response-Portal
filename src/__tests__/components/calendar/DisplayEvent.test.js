import React from 'react';
import DisplayEvent from "../../../components/calendar/DisplayEvent";
import TestRenderer from "react-test-renderer";
import {MemoryRouter as Router} from "react-router-dom";

let testRenderer;

beforeAll(() => {
  const event = {
    id: 0,
    title: 'Ã˜velse',
    start: new Date(2019, 0, 1, 19),
    end: new Date(2019, 0, 1, 20),
    participants: 'Alle',
  };
  const mockOnClick = jest.fn();
  testRenderer = TestRenderer.create(
    <Router>
      <DisplayEvent
        event={event}
        onChangeEvent={mockOnClick}
        onDeleteButtonClick={mockOnClick}
        onReviewButtonClick={mockOnClick}
      />
    </Router>
  );
});

test('renders correctly', () => {
  const tree = testRenderer.toJSON();
  expect(tree).toMatchSnapshot();
});
