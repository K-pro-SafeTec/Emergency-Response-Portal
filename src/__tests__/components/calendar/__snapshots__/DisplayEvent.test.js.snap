// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiPaper-root-1 MuiPaper-elevation2-5 MuiPaper-rounded-2 paper-small"
>
  <h1
    className="MuiTypography-root-28 MuiTypography-headline-33"
  >
    Ã˜velse
  </h1>
  <p
    className="MuiTypography-root-28 MuiTypography-body1-37"
  >
    19:00
    -
    20:00
  </p>
  <p
    className="MuiTypography-root-28 MuiTypography-body1-37"
  >
    Deltakere: 
    Alle
  </p>
  <button
    className="MuiButtonBase-root-89 MuiButton-root-63 MuiButton-outlined-71 MuiButton-outlinedPrimary-72"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "marginRight": "5px",
      }
    }
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiButton-label-64"
    >
      Endre
    </span>
    <span
      className="MuiTouchRipple-root-92"
    />
  </button>
  <button
    className="MuiButtonBase-root-89 MuiButton-root-63 MuiButton-outlined-71 MuiButton-outlinedSecondary-73"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiButton-label-64"
    >
      Slett
    </span>
    <span
      className="MuiTouchRipple-root-92"
    />
  </button>
</div>
`;
